### PR TITLE
Setup multi-architecture Docker image builds

### LINKED ISSUE #60: Multi Arch Images?
This image is very handy (thank you!) it currently supports `amd64` architecture. 

Would you be open to a PR that added support to the `publish-on-push-to-main` action that adds multi-arch support via the additional steps of:

https://github.com/docker/setup-qemu-action & https://github.com/docker/setup-buildx-action

and modification to the `build` script: e.g. 

```shell
docker buildx build --tag "$TAG" \
    --label org.opencontainers.image.created="$(date --rfc-3339=seconds)" \
    --label org.opencontainers.image.url="https://github.com/${GITHUB_REPOSITORY}" \
    --label org.opencontainers.image.documentation="https://github.com/octodns/octodns#readme" \
    --label org.opencontainers.image.source="https://github.com/${GITHUB_REPOSITORY}" \
    --label org.opencontainers.image.revision="$GITHUB_SHA" \
    --label org.opencontainers.image.licenses=MIT \
    --platform linux/amd64,linux/arm/v7,linux/arm64 \
    "$FLAVOR"
```

*Update:*

After some experimenting, I've found that this isn't going to work easily due to the decoupled `push` step and the `cibuild` / release as the images aren't loaded into the local store and there is some incompatibility around this: https://github.com/docker/roadmap/issues/371 (basically can't load the image into local store to test and/or push later if it's multi-arch so would require larger rework of this ci/cd pipeline or waiting until docker supports this, perhaps..)

### PR DESCRIPTION
**Pull request by parkr**
This provides optimized builds for arm-based platforms. This is based on @rjocoleman's work in #61.

Fixes #60.

### CODE REVIEW

File: script/generate-workflow-files
Comment (ross) [inline script/generate-workflow-files:57]: If this is going to stick around commented out it probably makes sense to add a comment as to why.

**Review by ross — APPROVED** [review]
👍 to the idea and implementation as it stands.

**Comment by parkr** [comment]
Tests are all passing, so I think this is good to go. The failing Publishing workflow runs are due to 429 from Docker.io...

**Comment by ross** [comment]
> The failing Publishing workflow runs are due to 429 from Docker.io...

I assume that's from all the CI runs/testing or something?

Otherwise 👍

File: script/generate-workflow-files
Comment (parkr) [reply script/generate-workflow-files]: Removed!

**Comment by parkr** [comment]
> I assume that's from all the CI runs/testing or something?

@ross Yes, exactly. It's because we log in before pulling/building (since the push happens in the same step) so that particular user is throttled. It looks like for the test workflow, the anonymous pulls are fine (no throttling by IP for the tests I've run here).

The test workflow builds all images on the 2 supported platforms (removed arm/v7 due to https://github.com/octodns/octodns-docker/issues/60#issuecomment-2341325973), so the only difference is that the Publish workflow actually `docker push`-es them, so I'm feeling confident that the `Publish` workflow should work as expected.

Either we wait to squash-merge until we get a positive Publish in this branch, or we can squash merge now and re-trigger the publish workflow later today (200 container image requests per six hours for free Docker accounts, from [docs](https://www.docker.com/increase-rate-limits/)). Up to you, either way I'd squash-merge here since my commits on this branch are not valuable to hold onto on `main` IMO.

**Comment by ross** [comment]
> Up to you, either way I'd squash-merge here since my commits on this branch are not valuable to hold onto on `main` IMO.

I'd probably wait until they pass and merge, but either works for me.

**Comment by parkr** [comment]
I'll squash my commits manually and force-push this evening to re-trigger the "Publish to Docker Hub" workflow after the rate limits reset!

**Comment by parkr** [comment]
We're green! Merging.

**Pull request merged**
